templates/nagios_plugin.py

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env python3
+
+# Copyright (C) 2019 Canonical Ltd.
+
+import nagios_plugin3
+import socket
+from subprocess import check_output
+
+snap_resources = ['kubectl', 'kube-apiserver', 'kube-controller-manager',
+                  'kube-scheduler', 'cdk-addons', 'kube-proxy']
+
+
+def check_snaps_installed():
+    """Confirm the snaps are installed, raise an error if not"""
+    for snap_name in snap_resources:
+        cmd = ['snap', 'list', snap_name]
+        try:
+            check_output(cmd).decode('UTF-8')
+        except Exception:
+            msg = '{} snap is not installed'.format(snap_name)
+            raise nagios_plugin3.CriticalError(msg)
+
+
+def test_connection(host, port):
+    try:
+        s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        s.settimeout(1)
+        s.connect((host, int(port)))
+        s.shutdown(socket.SHUT_RDWR)
+    finally:
+        s.close()
+
+
+def verify_local_connection_to_apiserver():
```

Comment:
Not sure we should include this since we'll likely be shutting down the insecure local port soon, to conform to the CIS benchmark for k8s. Check me on that @kwmonroe. I think the remote 6443 check is sufficient.

Reply:
Correct, it'll be a tech preview to disable 8080 for CIS in 1.17, with total removal of the insecure port in a subsequent release.